src/ui/components/SecondaryToolbox/react-devtools/components/SelectedElement.tsx

```diff
@@ -0,0 +1,247 @@
+import assert from "assert";
+import { ObjectId, PauseId } from "@replayio/protocol";
+import { FrontendBridge } from "@replayio/react-devtools-inline";
+import { MouseEvent, ReactNode, Suspense, useContext, useDeferredValue } from "react";
+
+import { selectNode } from "devtools/client/inspector/markup/actions/markup";
+import Expandable from "replay-next/components/Expandable";
+import Icon from "replay-next/components/Icon";
+import PropertiesRenderer from "replay-next/components/inspector/PropertiesRenderer";
+import Loader from "replay-next/components/Loader";
+import { InspectorContext } from "replay-next/src/contexts/InspectorContext";
+import { objectCache } from "replay-next/src/suspense/ObjectPreviews";
+import { suspendInParallel } from "replay-next/src/utils/suspense";
+import { ReplayClientContext } from "shared/client/ReplayClientContext";
+import { PreferencesKey } from "shared/user-data/LocalStorage/types";
+import useLocalStorageUserData from "shared/user-data/LocalStorage/useLocalStorageUserData";
+import { setSelectedPanel } from "ui/actions/layout";
+import { HooksRenderer } from "ui/components/SecondaryToolbox/react-devtools/components/HooksRenderer";
+import { nodesToFiberIdsCache } from "ui/components/SecondaryToolbox/react-devtools/injectReactDevtoolsBackend";
+import { ReactDevToolsListData } from "ui/components/SecondaryToolbox/react-devtools/ReactDevToolsListData";
+import {
+  ReplayWall,
+  StoreWithInternals,
+} from "ui/components/SecondaryToolbox/react-devtools/ReplayWall";
+import { inspectedElementCache } from "ui/components/SecondaryToolbox/react-devtools/suspense/inspectedElementCache";
+import { ReactElement } from "ui/components/SecondaryToolbox/react-devtools/types";
+import { useAppDispatch } from "ui/setup/hooks";
+
+import styles from "./SelectedElement.module.css";
+
+export function SelectedElement({
+  bridge,
+  element,
+  listData,
+  pauseId,
+  replayWall,
+  store,
+}: {
+  bridge: FrontendBridge;
+  element: ReactElement;
+  listData: ReactDevToolsListData;
+  pauseId: PauseId;
+  replayWall: ReplayWall;
+  store: StoreWithInternals;
+}) {
+  const replayClient = useContext(ReplayClientContext);
+  const { inspectFunctionDefinition } = useContext(InspectorContext);
+
+  const dispatch = useAppDispatch();
+
+  const deferredElement = useDeferredValue(element);
+
+  const { displayName, id } = deferredElement;
+
+  const isPending = element !== deferredElement;
+
+  const [inspectedElement, [, fiberIdsToNodeIds]] = suspendInParallel(
```

Comment:
:speech_balloon: yeah, this is what I was wondering about in the other file

Reply:
In this case it's different because we _can_ fetch these in parallel (we know enough up front to do that)